system/session/drivers/cookie.php

```diff
@@ -47,7 +48,7 @@ public function write($id, $cargo) {
 		$serialized = serialize($cargo);
 
 		// create a signature to verify content when unpacking
-		$sign = hash('md5', $serialized);
+		$sign = hash_hmac('sha256', $serialized, $session_key, true);
```

Comment:
Oh! Silly me, I didn't realise that this was in the cookies file! I understand it now. Cheers!